src/analysis/rust_type.rs

```diff
@@ -1,49 +1,152 @@
+use std::result;
+
 use library;
 
+pub struct Result(result::Result<String, String>);
```

Comment:
So it's a trade off between using the more verbose standard functions in place of `as_str` and wrapping values in this `Result` in 10 places?